test/saithrift/tests/sail2.py

```diff
@@ -757,3 +757,570 @@ def runTest(self):
+        vlan_member2 = sai_thrift_create_vlan_member(self.client, vlan_oid, port2, SAI_VLAN_TAGGING_MODE_UNTAGGED)
+        vlan_member3 = sai_thrift_create_vlan_member(self.client, vlan_oid, port3, SAI_VLAN_TAGGING_MODE_UNTAGGED)
+
+        ### Assign PVLAN ID with ports
+        attr_value = sai_thrift_attribute_value_t(u16=vlan_id)
+        attr = sai_thrift_attribute_t(id=SAI_PORT_ATTR_PORT_VLAN_ID, value=attr_value)
+        self.client.sai_thrift_set_port_attribute(port1, attr)
+        self.client.sai_thrift_set_port_attribute(port2, attr)
+        self.client.sai_thrift_set_port_attribute(port3, attr)
+
+        pkt1 = simple_tcp_packet(eth_dst=mac2,
+                                eth_src=mac1,
+                                ip_dst='192.168.0.2',
+                                ip_src='192.168.0.1',
+                                ip_id=105,
+                                ip_ttl=64)
+        exp_pkt1 = simple_tcp_packet(eth_dst=mac2,
+                                eth_src=mac1,
+                                ip_dst='192.168.0.2',
+                                ip_src='192.168.0.1',
+                                ip_id=105,
+                                ip_ttl=64)
+
+        pkt2 = simple_tcp_packet(eth_dst=mac1,
+                                eth_src=mac2,
+                                ip_dst='192.168.0.1',
+                                ip_src='192.168.0.2',
+                                ip_id=105,
+                                ip_ttl=64)
+        exp_pkt2 = simple_tcp_packet(eth_dst=mac1,
+                                eth_src=mac2,
+                                ip_dst='192.168.0.1',
+                                ip_src='192.168.0.2',
+                                ip_id=105,
+                                ip_ttl=64)
+        try:
+
+            ### Sending packet from Port1
+            send_packet(self, 0, str(pkt1))
+
+            ### verify the packet @ Port2
+            verify_packets(self, exp_pkt1, [1, 2])
+
+            ### Sending packet from Port2
+            send_packet(self, 1, str(pkt2))
+
+            ### verify the packet @ Port1
+            verify_packets(self, exp_pkt2, [0])
+
+            time.sleep(1)
+
+            ### Sending packet from Port1
+            send_packet(self, 0, str(pkt1))
+
+            ### verify the packet @ Port3
+            verify_packets(self, exp_pkt1, [1])
+
+            ### verify the packet @ Port3
+            verify_no_packet(self, exp_pkt1, 2)
+
+            time.sleep(1)
+
+            ### Send packet (src_mac=MAC2 and dst_mac= MAC1) on port3
+            send_packet(self, 2, str(pkt2))
+
+            ### verify the packet @ Port1
+            verify_packets(self, exp_pkt2, [0])
+
+            time.sleep(1)
+
+            ### Send packet (src_mac=MAC1 and dst_mac=MAC2) from port 1
+            send_packet(self, 0, str(pkt1))
+
+            ### verify the packet @ Port3 & Port2
```

Comment:
verify the packet @Port 3 and no packet at @Port2

Reply:
corrected description